File: app/src/main/java/com/dumuzeyn/mp3player/AppearanceState.java
package com.dumuzeyn.mp3player;

/** Persisted visual and interaction preferences. */
final class AppearanceState {
    int customTimerMinutes = 10;
    int resumeWindowMinutes = 120;
    int particleFrequency = 45;
    int particleSize = 100;
    int particleLifetime = 100;
    int particlePrimaryColor;
    int particleSecondaryColor;
    int fullPlayerRotationSpeed = 100;
    int cardOpacity = 82;
    int songCardOpacity = 82;
    int favoriteCardOpacity = 82;
    int playlistCardOpacity = 82;
    int genreCardOpacity = 82;
    int artistCardOpacity = 82;
    int albumCardOpacity = 82;
    int settingsCardOpacity = 82;
    int miniPlayerCardOpacity = 82;
    int headerCardOpacity = 82;
    int dialogCardOpacity = 82;
    boolean dark;
    boolean animations = true;
    boolean particlesEnabled = true;
    int playerBackgroundMode = BackgroundSettingsController.MODE_GRADIENT;
    int mainBackgroundMode = BackgroundSettingsController.MODE_SOLID;
    int mainSolidBackground;
    int playerSolidBackground;
    String mainBackgroundMediaUri = "";
    String playerBackgroundMediaUri = "";
    int mainBackgroundBlur = 20;
    int playerBackgroundBlur = 20;
    boolean circularCovers;
    int mainGradientStart = 0xff351b5d;
    int mainGradientEnd = 0xff3a3013;
    int playerGradientStart = 0xff351b5d;
    int playerGradientEnd = 0xff3a3013;
    String language = "ru";
    String themeMode = "light";
    int customBg = -1;
    int customFg = -16777216;
    int customSecondaryAccent = 0xffffd000;
    int customTextColor;
    boolean textOutlineEnabled;
    int textOutlineColor;
}

File: app/src/main/java/com/dumuzeyn/mp3player/AppearanceState.kt
package com.dumuzeyn.mp3player

/** Persisted visual and interaction preferences. */
class AppearanceState {
    @JvmField var customTimerMinutes = 10
    @JvmField var resumeWindowMinutes = 120
    @JvmField var particleFrequency = 45
    @JvmField var particleSize = 100
    @JvmField var particleLifetime = 100
    @JvmField var particlePrimaryColor = 0
    @JvmField var particleSecondaryColor = 0
    @JvmField var fullPlayerRotationSpeed = 100
    @JvmField var cardOpacity = 82
    @JvmField var songCardOpacity = 82
    @JvmField var favoriteCardOpacity = 82
    @JvmField var playlistCardOpacity = 82
    @JvmField var genreCardOpacity = 82
    @JvmField var artistCardOpacity = 82
    @JvmField var albumCardOpacity = 82
    @JvmField var settingsCardOpacity = 82
    @JvmField var miniPlayerCardOpacity = 82
    @JvmField var headerCardOpacity = 82
    @JvmField var dialogCardOpacity = 82
    @JvmField var dark = false
    @JvmField var animations = true
    @JvmField var particlesEnabled = false
    @JvmField var playerBackgroundMode = BackgroundSettingsController.MODE_GRADIENT
    @JvmField var mainBackgroundMode = BackgroundSettingsController.MODE_GRADIENT
    @JvmField var mainSolidBackground = 0
    @JvmField var playerSolidBackground = 0
    @JvmField var mainBackgroundMediaUri = ""
    @JvmField var playerBackgroundMediaUri = ""
    @JvmField var mainBackgroundBlur = 20
    @JvmField var playerBackgroundBlur = 20
    @JvmField var circularCovers = false
    @JvmField var mainGradientStart = VoltunePalette.GRADIENT_BLUE
    @JvmField var mainGradientEnd = VoltunePalette.GRADIENT_PURPLE
    @JvmField var playerGradientStart = VoltunePalette.GRADIENT_BLUE
    @JvmField var playerGradientEnd = VoltunePalette.GRADIENT_PURPLE
    @JvmField var language = "ru"
    @JvmField var themeMode = "light"
    @JvmField var customBg = -1
    @JvmField var customFg = -16777216
    @JvmField var customSecondaryAccent = VoltunePalette.GOLD
    @JvmField var customTextColor = 0
    @JvmField var textOutlineEnabled = false
    @JvmField var textOutlineColor = 0
}
